src/EditorFeatures/Core/Implementation/AutomaticCompletion/BraceCompletionSessionProvider.BraceCompletionSession.cs

```diff
@@ -364,6 +352,61 @@ private void MoveCaretToClosingPoint()
                 }
             }
 
+            private void ApplyBraceCompletionResult(BraceCompletionResult result)
+            {
+                using var edit = SubjectBuffer.CreateEdit();
+                foreach (var change in result.TextChanges)
+                {
+                    edit.Replace(change.Span.ToSpan(), change.NewText);
+                }
+
+                edit.ApplyAndLogExceptions();
+
+                Debug.Assert(SubjectBuffer.CurrentSnapshot[OpeningPoint.GetPosition(SubjectBuffer.CurrentSnapshot)] == OpeningBrace,
+                "The opening point does not match the opening brace character");
+                Debug.Assert(SubjectBuffer.CurrentSnapshot[ClosingPoint.GetPosition(SubjectBuffer.CurrentSnapshot) - 1] == ClosingBrace,
+                "The closing point does not match the closing brace character");
+
+                var snapshotPoint = SubjectBuffer.CurrentSnapshot.GetPoint(result.CaretLocation);
+                var line = snapshotPoint.GetContainingLine();
+
+                if (line.GetText().IsNullOrWhiteSpace())
+                {
+                    // If the caret is on an empty line, put it at the correct indentation using virtual space.
+                    PutCaretOnLine(this, _smartIndentationService, line.LineNumber);
+                }
+                else
+                {
+                    // Not on an empty line, just put the caret at the specified location.
+                    TextView.TryMoveCaretToAndEnsureVisible(snapshotPoint);
+                }
```

Comment:
```suggestion
                }
                
                return;
```

placing an explicit return before the local functions makes it immediately obvious that hte main 'code' of hte method is done and the locals are not interspersed with it.  without this, one has to scan all the way to the end to see if there's any more interesting code that is executed.